src/main/java/uk/ac/ebi/ena/readtools/loader/common/converter/AbstractReadConverter.java

```diff
@@ -87,14 +85,13 @@ public final void run() {
 
             do {
                 synchronized (readWriter) {
-                    for (int yield = YIELD_CYCLES; yield > 0; --yield)
+                    for (int yield = YIELD_CYCLES; yield > 0 && keepRunning.get(); --yield)
```

Comment:
for (int readCount = ERROR_CHECK_BATCH_SIZE; readCount > 0 ...

keepRunning  -> isWithinReadLimit